### PR TITLE
Jetpack Connect: create a core Jetpack remote install flow and handle on success requests.

### PR DESCRIPTION
**Pull request by AnnaMag**
Flow starts at `/jetpack/connect` and redirects to `jetpack/connect/install` where the credentials form lives for .org sites with Jetpack not installed or activated. Uses a dedicated wpcom API endpoint
and selectors defined in https://github.com/Automattic/wp-calypso/pull/22555.

*Changes included:*
* adds feature flag, route definition for `jetpack/connect/install`,
* base UI for the credentials ( additional styling in the follow-up),
* handles **on success** responses ( valid credentials and admin permissions for the site),

*Not covered:*
- full styling for entire flow ( Worg logo in the credentials form, footer links from the mockups, new button branding, wizard counter ),

**To test:**
- Apply D10065 on the .com sandbox and sandbox.
- Checkout this branch on local Calypso and start at `calypso.localhost:3000/jetpack/connect/`
Input url of a remote site with:
1. Jetpack installed and activated- the flow should redirect to the JPC authorization form,
2. no Jetpack installed (or installed but not activated)- verify the redirect to the screen with form credentials at `jetpack/connect/install`,

- input valid site credentials,
- verify that Jetpack was installed (or activated if already installed) and the flow redirects to the JPC authorization step ( .com login and/or JP plans page).

screen shot 2018-02-26 at 22 13 05

### CODE REVIEW

File: config/development.json [hunk 1]
```diff
@@ -44,7 +44,6 @@
 		"devdocs/redirect-loggedout-homepage": true,
 		"devdocs/components-usage-stats": false,
 		"dev/test-helper": true,
-		"dev/preferences-helper": true,
```
Comment (seear) [inline config/development.json]: Probably this line should not be touched?

Comment (AnnaMag) [reply config/development.json]: bad rebase- removed 👍

File: client/jetpack-connect/remote-credentials.js [hunk 2]
```diff
@@ -0,0 +1,227 @@
...
+		};
+	}
+
+	handleSubmit = event => {
+		const { siteToConnect } = this.props;
+		event.preventDefault();
+
+		const url = siteToConnect;
+		const username = this.state.username;
+		const password = this.state.password;
```
Comment (seear) [inline client/jetpack-connect/remote-credentials.js]: These `const`s don't seem necessary?

File: client/jetpack-connect/remote-credentials.js [hunk 3]
```diff
@@ -0,0 +1,227 @@
...
+		}
+		if ( installError ) {
+			//handle errors
+		}
+	}
+
+	getChangeHandler = field => event => {
+		this.setState( { [ field ]: event.target.value } );
+	};
+	getHeaderImage() {
```
Comment (seear) [inline client/jetpack-connect/remote-credentials.js]: Missing blank line

File: client/jetpack-connect/remote-credentials.js [hunk 4]
```diff
@@ -0,0 +1,227 @@
...
+		const { translate } = this.props;
+
+		return translate( 'Add your website credentials' );
+	}
+
+	getSubHeaderText() {
+		const { translate } = this.props;
+
+		return (
+			<div className="jetpack-connect__install-step">
```
Comment (seear) [inline client/jetpack-connect/remote-credentials.js]: This `<div>` gives a warning:

screen shot 2018-02-27 at 12 58 04

A `<span>` should be ok if we need the styling.

File: client/jetpack-connect/style.scss [hunk 5]
```diff
@@ -583,3 +583,47 @@
...
+
+.jetpack-connect__password-form {
+	position: relative;
+
+	.gridicon {
+		position: absolute;
+		z-index: z-index('root', '.jetpack-connect__password-form .gridicon');
+		left: 8px;
+		top: 7px;
+		fill: $gray;
```
Comment (seear) [inline client/jetpack-connect/style.scss]: `$gray-lighten-20` to match the _username_ gridicon.

**Pull request merged**